usecase/usecase.go

```diff
@@ -0,0 +1,65 @@
+package usecase
+
+///cambir por use case
+import (
+	"errors"
+
+	"github.com/rayl17/academy-go-q42021/model"
+	service "github.com/rayl17/academy-go-q42021/service/csv"
+)
+
+type Usecaseinterface interface {
+	GetPokemonByName(name string) (*model.Pokemon, error)
+	GetPokemonByID(id string) (*model.Pokemon, error)
+	SavePokemonByNameAPI(name string) (string, error)
+}
+
+type PokemonUsecase struct {
+	service service.ServiceInterface
+}
+
+func NewUseCase(ps service.ServiceInterface) *PokemonUsecase {
+
+	return &PokemonUsecase{
+		service: ps,
+	}
+
+}
+
+func (u *PokemonUsecase) GetPokemonByName(name string) (*model.Pokemon, error) {
+
+	pokemon, err := u.service.GetPokemonByName(name)
+
+	if err != nil {
+
+		return nil, errors.New("Pokemon not found")
+	}
+
+	return pokemon, nil
+
+}
+
+// int
+func (u *PokemonUsecase) GetPokemonByID(id string) (*model.Pokemon, error) {
+
+	pokemon, err := u.service.GetPokemonByID(id)
+
+	if err != nil {
+
+		return nil, errors.New("Pokemon not found")
```

Comment:
Do not dynamically create errors, remember to define them inside the current usecase layer, in an errors package.

Also, you should wrap the error if something happens; when back-propagating the error, you could get chained errors, and debugging will be easier by having these wrapped errors.

Fix this for this and all other returned errors.